pkg/function/kube_task.go

```diff
@@ -83,12 +89,55 @@ func kubeTaskPodFunc(cli kubernetes.Interface) func(ctx context.Context, pod *v1
 		}
 		// Wait for pod completion
 		if err := kube.WaitForPodCompletion(ctx, cli, pod.Namespace, pod.Name); err != nil {
-			return nil, errors.Wrapf(err, "Failed while waiting for Pod %s to complete", pod.Name)
+			logs, logerr := recentLogsAsOutput(ctx, cli, pod.Namespace, pod.Name, kubeTaskLogsOutputKey)
+			if logerr != nil {
+				return nil, logerr
+			}
+
+			return map[string]interface{}{
+				logs.Key: logs.Value,
+			}, errors.Wrapf(err, "Failed while waiting for Pod %s to complete", pod.Name)
 		}
 		return out, err
 	}
 }
 
+// recentLogsAsOutput creates an artifact output containing the recent logs of
+// a pod. The output maps 'key' to the log string.
+// The number of log lines to retrieve is determined by either the value of the
+// TaskLogsTailSizeEnv environment variable, or the default
+// kube.TaskLogsTailSizeDefault.
+func recentLogsAsOutput(ctx context.Context, cli kubernetes.Interface, namespace, pod, key string) (*output.Output, error) {
+	var tailSize int64
+	tailSizeEnv, exists := os.LookupEnv(kubeTaskLogsTailSizeEnv)
+	if exists {
+		parsed, err := strconv.ParseInt(tailSizeEnv, 10, 0)
+		if err != nil {
+			return nil, err
+		}
+		tailSize = parsed
+	}
+
+	if tailSize == 0 {
+		tailSize = kube.TaskLogsTailSizeDefault
```

Comment:
I was thinking are there chances that the error that we are looking for is not actually available in last 15 lines? Do we rely to setting env var in that case?
And if we want to set the env var, I think figuring out how many lines do we want to look for is a bit challenging. How would we decide this number?

Reply:
We won't know ahead of time. Hence, I am thinking of making it configurable. We definitely don't want to blindly pull down all the logs, because that will get saved into etcd.

Reply:
right. But I was thinking even if its configurable how can someone decide the number of lines we want to tail from the pod. That's a bit challenging in my opinion.